File: frontend/src/api/documentSummary.ts
import { apiRequest } from "./http";
import { downloadExport, type ExportFormat } from "@/utils/download";

export interface DocumentSummarySource {
  pageNumber?: number;
  chunkIndex: number;
  excerpt: string;
}

export interface DocumentSummary {
  id: number;
  documentId: number;
  projectId: number;
  creatorId: number;
  creatorName?: string;
  documentName: string;
  title: string;
  content: string;
  sources: DocumentSummarySource[];
  model?: string;
  createdAt: string;
  updatedAt: string;
}

export const documentSummaryApi = {
  generate: (documentId: number, title?: string) => apiRequest<DocumentSummary>({
    url: `/documents/${documentId}/ai/summary`, method: "POST", data: { title }, timeout: 300000,
  }),
  list: (documentId: number) =>
    apiRequest<DocumentSummary[]>({ url: `/documents/${documentId}/ai/summaries` }),
  detail: (summaryId: number) =>
    apiRequest<DocumentSummary>({ url: `/ai/document-summaries/${summaryId}` }),
  update: (summaryId: number, data: { title: string; content: string }) =>
    apiRequest<DocumentSummary>({ url: `/ai/document-summaries/${summaryId}`, method: "PUT", data }),
  remove: (summaryId: number) =>
    apiRequest<void>({ url: `/ai/document-summaries/${summaryId}`, method: "DELETE" }),
  export: (summaryId: number, title: string, format: ExportFormat) =>
    downloadExport(`/ai/document-summaries/${summaryId}/export`, title, format),
};

File: ai-service/app/models/document_summary.py
from pydantic import BaseModel, Field


class DocumentSummaryChunk(BaseModel):
    page_number: int | None = Field(default=None, alias="pageNumber")
    chunk_index: int = Field(alias="chunkIndex", ge=0)
    content: str = Field(min_length=1)


class DocumentSummaryRequest(BaseModel):
    document_id: int = Field(alias="documentId", gt=0)
    project_id: int = Field(alias="projectId", gt=0)
    user_id: int = Field(alias="userId", gt=0)
    document_name: str = Field(alias="documentName", min_length=1)
    file_type: str = Field(alias="fileType")
    chunks: list[DocumentSummaryChunk] = Field(min_length=1)


class DocumentSummarySource(BaseModel):
    page_number: int | None = Field(default=None, alias="pageNumber")
    chunk_index: int = Field(alias="chunkIndex")
    excerpt: str

    model_config = {"populate_by_name": True}


class DocumentSummaryResponse(BaseModel):
    content: str
    model: str
    sources: list[DocumentSummarySource] = Field(default_factory=list)
